crates/common/tedge_config/src/tedge_config_cli/tedge_config_repository.rs

```diff
@@ -2,6 +2,42 @@ use crate::*;
 use std::fs;
 use std::path::PathBuf;
 use tedge_utils::fs::atomically_write_file_sync;
+use tracing::warn;
+
+/// Displays warning messages for any unknown toml **fields and or keys**.
+///
+/// It does **not** display values.
+macro_rules! display_unknown_for {
```

Comment:
Can this macro return a vector of messages instead of calling implicitly `tracing::warn` ?